akka-http-core/src/main/scala/akka/http/impl/engine/ws/WebSocketClientBlueprint.scala

```diff
@@ -57,68 +57,82 @@ object WebSocketClientBlueprint {
     val renderedInitialRequest =
       HttpRequestRendererFactory.renderStrict(RequestRenderingContext(initialRequest, hostHeader), settings, log)
 
-    class UpgradeStage extends StatefulStage[ByteString, ByteString] {
-      type State = StageState[ByteString, ByteString]
-
-      def initial: State = parsingResponse
-
-      def parsingResponse: State = new State {
-        // a special version of the parser which only parses one message and then reports the remaining data
-        // if some is available
-        val parser = new HttpResponseParser(settings.parserSettings, HttpHeaderParser(settings.parserSettings)()) {
-          var first = true
-          override def handleInformationalResponses = false
-          override protected def parseMessage(input: ByteString, offset: Int): StateResult = {
-            if (first) {
-              first = false
-              super.parseMessage(input, offset)
-            } else {
-              emit(RemainingBytes(input.drop(offset)))
-              terminate()
+    //from here
+    class UpgradeStage extends GraphStage[FlowShape[ByteString, ByteString]] {
+
+      val out: Outlet[ByteString] = Outlet("UpgradeStage.out")
+      val in: Inlet[ByteString] = Inlet("UpgradeStage.in")
+
+      override val shape: FlowShape[ByteString, ByteString] = FlowShape(in, out)
+
+      override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+        new GraphStageLogic(shape) with InHandler with OutHandler {
+
+          def initial: InHandler = parsingResponse
```

Comment:
initial does nothing in `GraphStage`, you have to set the in handler, like this instead: `setHandler(in, parsingStage)`